x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.ts

```diff
@@ -5,40 +5,39 @@
  */
 
 import Hapi from 'hapi';
-import { chunk, isEmpty, isFunction } from 'lodash/fp';
+import { chunk, isEmpty } from 'lodash/fp';
 import { extname } from 'path';
 import uuid from 'uuid';
+
 import { createPromiseFromStreams } from '../../../../../../../../../src/legacy/utils/streams';
 import { DETECTION_ENGINE_RULES_URL } from '../../../../../common/constants';
+import { LegacyServices, LegacyRequest } from '../../../../types';
 import { createRules } from '../../rules/create_rules';
 import { ImportRulesRequest } from '../../rules/types';
-import { ServerFacade } from '../../../../types';
 import { readRules } from '../../rules/read_rules';
 import { getIndexExists } from '../../index/get_index_exists';
-import {
-  callWithRequestFactory,
-  getIndex,
-  createBulkErrorObject,
-  ImportRuleResponse,
-} from '../utils';
+import { getIndex, createBulkErrorObject, ImportRuleResponse } from '../utils';
 import { createRulesStreamFromNdJson } from '../../rules/create_rules_stream_from_ndjson';
 import { ImportRuleAlertRest } from '../../types';
 import { patchRules } from '../../rules/patch_rules';
 import { importRulesQuerySchema, importRulesPayloadSchema } from '../schemas/import_rules_schema';
-import { KibanaRequest } from '../../../../../../../../../src/core/server';
+import { GetScopedClients } from '../../../../services';
 
 type PromiseFromStreams = ImportRuleAlertRest | Error;
 
 const CHUNK_PARSED_OBJECT_SIZE = 10;
 
-export const createImportRulesRoute = (server: ServerFacade): Hapi.ServerRoute => {
+export const createImportRulesRoute = (
+  config: LegacyServices['config'],
+  getClients: GetScopedClients
+): Hapi.ServerRoute => {
   return {
     method: 'POST',
     path: `${DETECTION_ENGINE_RULES_URL}/_import`,
     options: {
       tags: ['access:siem'],
       payload: {
-        maxBytes: server.config().get('savedObjects.maxImportPayloadBytes'),
+        maxBytes: config().get('savedObjects.maxImportPayloadBytes'),
```

Comment:
Happy we can still use the `savedObjects` key here for now rather than making another one. Otherwise I don't think its harmful if we had to make our own key at some point.

The hope is they don't hit these limits and ask about it right now and they're really to prevent a memory overrun issue. We don't do true streaming of data but rather "sink" it all here at the moment due to some technical issues.

Reply:
We're not able to access this config on new platform, and we will in fact have to add our own config for these values (unless savedObjects makes them available to us via their contract). I punted on that for this PR since there's no shim available for the NP config stuff.